linux/src/async_socket_linux.c

```diff
@@ -0,0 +1,695 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+
+#include <stdlib.h>
+#include <inttypes.h>
+#include <sys/epoll.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include <arpa/inet.h>
+#include <sys/epoll.h>
+#include <sys/socket.h>
+
+#include "macro_utils/macro_utils.h"
+
+#include "c_logging/xlogging.h"
+
+#include "c_pal/gballoc_hl.h"
+#include "c_pal/gballoc_hl_redirect.h"
+#include "c_pal/execution_engine.h"
+#include "c_pal/interlocked.h"
+#include "c_pal/sync.h"
+#include "c_pal/socket_handle.h"
+#include "c_pal/threadapi.h"
+
+#include "c_pal/async_socket.h"
+
+#define MAX_EVENTS_NUM      64
```

Comment:
erm, does this mean we can only have 64 sockets?
That will not really work, will it?

Reply:
No this means that the on older system (This has been fixed with new kernel, the doc didn't say which kernel version has been fixed) this value is the number of spots in the array it starts with.  It grows as needed.